web-services/security/src/main/java/datawave/security/authorization/remote/RemoteUserOperationsImpl.java

```diff
@@ -0,0 +1,279 @@
+        }, () -> suffix);
+        
+    }
+    
+    public <T> T readResponse(HttpEntity entity, ObjectReader reader) throws IOException {
+        if (entity == null) {
+            return null;
+        } else {
+            String content = getContent(entity.getContent());
+            try {
+                return reader.readValue(content);
+            } catch (IOException ioe) {
+                log.error("Failed to read entity content.  Trying as a VoidResponse.", ioe);
+                log.error(content);
+                VoidResponse response = voidResponseReader.readValue(content);
+                throw new RuntimeException(response.getMessages().toString());
+            }
+        }
+    }
+    
+    private String getContent(InputStream content) throws IOException {
+        StringBuilder builder = new StringBuilder();
+        InputStreamReader reader = new InputStreamReader(content, "UTF8");
+        char[] buffer = new char[1024];
+        int chars = reader.read(buffer);
+        while (chars >= 0) {
+            builder.append(buffer, 0, chars);
+            chars = reader.read(buffer);
+        }
+        return builder.toString();
+    }
+    
+    public static String getEntities(Object callerObject) {
+        if (callerObject instanceof DatawavePrincipal) {
+            DatawavePrincipal callerPrincipal = (DatawavePrincipal) callerObject;
+            return callerPrincipal.getProxiedUsers().stream().map(u -> new StringBuilder().append('<').append(u.getDn().subjectDN()).append('>'))
+                            .collect(Collectors.joining());
+        } else {
+            throw new RuntimeException("Cannot use " + callerObject.getClass().getName() + " as a caller object");
+        }
+    }
+    
+    public static String getIssuers(Object callerObject) {
+        if (callerObject instanceof DatawavePrincipal) {
+            DatawavePrincipal callerPrincipal = (DatawavePrincipal) callerObject;
+            return callerPrincipal.getProxiedUsers().stream().map(u -> new StringBuilder().append('<').append(u.getDn().issuerDN()).append('>'))
+                            .collect(Collectors.joining());
+        } else {
+            throw new RuntimeException("Cannot use " + callerObject.getClass().getName() + " as a caller object");
+        }
+    }
+    
+    private <T> T executeGetMethodWithRuntimeException(String uriSuffix, Consumer<URIBuilder> uriCustomizer, Consumer<HttpGet> requestCustomizer,
+                    IOFunction<T> resultConverter, Supplier<String> errorSupplier) {
+        init();
+        try {
+            return executeGetMethod(uriSuffix, uriCustomizer, requestCustomizer, resultConverter, errorSupplier);
+        } catch (URISyntaxException e) {
+            throw new RuntimeException("Invalid URI: " + e.getMessage(), e);
+        } catch (IOException e) {
+            config.getFailureCounter().inc();
+            throw new RuntimeException(e.getMessage(), e);
+        }
+    }
+    
+    @Override
```

Comment:
It makes me sad too.  We are fixing this in the query microservices to be a configuration object instead of a mess of setters.  Can we change this later?